### ISSUE TITLE
any clue about what the moving average

### ISSUE THREAD

**Issue by vince62s** [marian-nmt/marian#246]
brings in terms of BLEU on a standard run / dataset for the base Transformer ?

**Comment by emjotde** [comment]
It seems to make checkpoint averaging unnecessary (being basically a continuous checkpoint average), so seems to give you about as much. I also found it to make training much more stable and repeatable. 

From my experience it usually gives you better than the max over possible checkpoints, which is quite desirable behavior. 

If you implement that or use, make sure to cite Marian paper. I am unaware of anyone else using this for NMT :)

**Comment by vince62s** [comment]
it's always my pleasure to cite you :) , but I read also this in http://www.statmt.org/wmt18/pdf/WMT035.pdf where they cite (Zhang et al., 2015)
is it different ?

**Comment by emjotde** [comment]
Yes and no. The math for the parameter update is the same, but their average has a tiny window and decays very quickly. Also this is for SGD not ADAM. And the exact implementation from that paper does not work at all, tried it :) 

Their goal is to make multi-gpu training for async SGD work, and it fails at that, at least for NMT.  

My window is huge, I think by default the last 100,000 iterations still have a 1% influence on the current model. 

You can cite both. If I ever were to describe that method I would cite Zhang et al. as but different in function and goal. For me it's really the smoothing effect I want.

**Comment by vince62s** [comment]
I was intrigued by the synchronous-SGD expression in Alibaba's paper, I had never seen anyone training the Transformer with SGD (I even believed it was just a qualificative of synchronous).
On the other end they get very very good results on EN<>RU.

**Comment by emjotde** [comment]
I think they only used SGD for fine-tuning if I am parsing this correctly. And they surely did not use Elastic SGD which is the method proposed in the Zhang et al. 2014 paper.

**Comment by emjotde** [comment]
That's still fine. The updates are of course weighted and current updates weigh much heavier. I have yet see a situation where it would not be beneficial.

**Comment by emjotde** [comment]
And the window get's modified by the parameter, the larger the parameter the smaller the window. I use 0.0001 by default, 0.001 will be a much shorter window.

**Comment by tomsbergmanis** [comment]
@emjotde Is there an implementation to what you describe above?

**Comment by snukky** [comment]
It's `--exponential-smoothing FLOAT`.
